src/app/backend/resource/secret/secretdetail.go

```diff
@@ -51,11 +52,24 @@ func GetSecretDetail(client *client.Client, namespace, name string) (*SecretDeta
 	return getSecretDetail(rawSecret), nil
 }
 
+func hideSecretData(rawSecret *api.Secret) map[string][]byte {
+	// For opaque and basicAuth secret types the credentials like username, password and etc. should be hidden.
+	if rawSecret.Type == api.SecretTypeOpaque || rawSecret.Type == api.SecretTypeBasicAuth {
+		data := make(map[string][]byte)
+		for key, value := range rawSecret.Data {
+			data[key] = []byte(fmt.Sprintf("%d bytes", len(value)))
```

Comment:
@bryk @maciaszczykm  personally unless it is a startup flag for the dashboard I don't think there should be an option on a production kubernetes dashboard to see opaque secret, as this can expose a lot of things the general kubernetes admin/user thought was secure to anyone with access to the UI.